.github/workflows/tests.yml

```diff
@@ -4,40 +4,87 @@ on:
     branches: master
   pull_request: []
 jobs:
-  tests:
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        os: [ubuntu-latest, macOS-latest]
+  lint:
+    name: Lint Dockerfile
+    runs-on: ubuntu-latest
     steps:
     - name: Set up Git repository
       uses: actions/checkout@master
+    - name: Check Dockerfile
+      run: |
+        docker run --rm -i hadolint/hadolint < Dockerfile
 
+  test-mac:
+    name: Build [macOS]
+    runs-on: macOS-latest
+    steps:
+    - name: Set up Git repository
+      uses: actions/checkout@master
     - name: Set up Homebrew
       run: |
         brew update-reset /usr/local/Homebrew
         ln -s "$PWD" "/usr/local/Homebrew/Library/Taps/homebrew/homebrew-portable-ruby"
-      if: matrix.os == 'macOS-latest'
-
-    - name: Check Dockerfile
+    - name: Build Portable Ruby
       run: |
-        sudo npm install -g dockerfilelint
-        dockerfilelint docker/Dockerfile.*
-      if: matrix.os == 'ubuntu-latest'
-
-    - name: Run brew style
+        mkdir -p bottle/
+        cd bottle
+        brew portable-package portable-ruby
+    - name: Upload Portable Ruby
+      uses: actions/upload-artifact@master
+      with:
+        name: portable-ruby-mac
+        path: bottle
+    - name: Test Portable Ruby
       run: |
-        brew style homebrew/portable-ruby
-      if: matrix.os == 'macOS-latest'
+        mkdir -p portable-ruby/
+        tar --strip-components 2 -C portable-ruby -xf bottle/portable-ruby--*.tar.gz
+        export HOMEBREW_DEVELOPER=1
+        export HOMEBREW_RUBY_PATH=$PWD/portable-ruby/bin/ruby
+        export HOMEBREW_LANGUAGES=en-US
+        brew config
+        brew tests
 
+  test-linux:
+    name: Build [Linux ${{ matrix.arch.name }}]
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        arch:
+          - { name: x86_64, platform: linux/amd64, build_img: 'debian/eol:wheezy', test_img: 'ubuntu:xenial' }
+          - { name: arm32v6, platform: linux/arm/v6, build_img: 'resin/rpi-raspbian:wheezy', test_img: 'ubuntu:xenial' }
+          - { name: arm64v8, platform: linux/arm64/v8, build_img: 'debian/eol:jessie', test_img: 'ubuntu:xenial' }
```

Comment:
if the `test_img` is always the same (or at least the same for all three): can it be hardcoded somewhere it's shared between all three?